plugins/inputs/sflow/metricencoder.go

```diff
@@ -0,0 +1,46 @@
+package sflow
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/inputs/sflow/types"
+)
+
+func makeMetrics(p *types.V5Format) ([]telegraf.Metric, error) {
+	metrics := []telegraf.Metric{}
```

Comment:
Initialize to len(FlowRecords)?

Reply:
You'd have to loop samples first and count it up, then do the allocation. I'm not entirely sure it's worth it.